packages/terminal/src/widget.ts

```diff
@@ -271,39 +293,44 @@ export class Terminal extends Widget {
    * Resize the terminal based on computed geometry.
    */
   private _resizeTerminal() {
-    fit(this._term);
-    if (this._offsetWidth === -1) {
-      this._offsetWidth = this.node.offsetWidth;
-    }
-    if (this._offsetHeight === -1) {
-      this._offsetHeight = this.node.offsetHeight;
-    }
-    this._setSessionSize();
-    this._needsResize = false;
+    this._ready.promise.then(() => {
```

Comment:
It doesn't matter that much since it's a private function, but we could return the ready promise here so that the resize can be awaited if the user wants to.

Reply:
which raises the larger question of whether `ready.promise` needs to be exposed publicly...
```ts

  /**
   * A promise that resolves when the terminal is ready.
   */
  get ready(): Promise<void> {
    return this._ready.promise;
  }
```